master/buildbot/test/unit/test_db_builds.py

```diff
@@ -496,3 +499,37 @@ def test_results_are_correctly_filtered_for_user(self):
             self._collect_ids(second_user_builds, 'builds_id'),
             range(first_user_builds_count, second_user_max_build_id)
         )
+
+    @defer.inlineCallbacks
+    def test_results_has_branch_and_codebase(self):
+        day_count = 7
+        example_data = [
+            fakedb.SourceStamp(id=1, branch="HEAD", sourcestampsetid=2, codebase="fmod"),
+            fakedb.SourceStamp(id=2, branch="trunk", sourcestampsetid=3, codebase="bar"),
```

Comment:
Make sure to add a test for when there are multiple sourcestamps attached to a single `sourcestampsetid`